ui/js/component/publishForm/view.jsx

```diff
@@ -0,0 +1,924 @@
+    });
+  }
+
+  handleOtherLicenseUrlChange(event) {
+    this.setState({
+      otherLicenseUrl: event.target.value,
+    });
+  }
+
+  handleChannelChange(channelName) {
+    this.setState({
+      channel: channelName,
+    });
+    const nameChanged = () => this.nameChanged(this.state.rawName);
+    setTimeout(nameChanged.bind(this), 500, { once: true });
+  }
+
+  handleTOSChange(event) {
+    this.setState({
+      tosAgree: event.target.checked,
+    });
+  }
+
+  handleCreateChannelClick(event) {
+    if (this.state.newChannelName.length < 5) {
+      this.refs.newChannelName.showError(
+        __("LBRY channel names must be at least 4 characters in length.")
+      );
+      return;
+    }
+
+    this.setState({
+      creatingChannel: true,
+    });
+
+    const newChannelName = this.state.newChannelName;
+    lbry
+      .channel_new({
+        channel_name: newChannelName,
+        amount: parseFloat(this.state.newChannelBid),
+      })
+      .then(
+        () => {
+          setTimeout(() => {
+            this.setState({
+              creatingChannel: false,
+            });
+
+            this._updateChannelList(newChannelName);
+          }, 10000);
+        },
+        error => {
+          // TODO: better error handling
+          this.refs.newChannelName.showError(
+            __("Unable to create channel due to an internal error.")
+          );
+          this.setState({
+            creatingChannel: false,
+          });
+        }
+      );
+  }
+
+  getLicense() {
+    switch (this.state.licenseType) {
+      case "copyright":
+        return this.state.copyrightNotice;
+      case "other":
+        return this.state.otherLicenseDescription;
+      default:
+        return this._meta_license.getSelectedElement().text;
+    }
+  }
+
+  getLicenseUrl() {
+    switch (this.state.licenseType) {
+      case "copyright":
+        return "";
+      case "other":
+        return this.state.otherLicenseUrl;
+      default:
+        return this._meta_license.getSelectedElement().getAttribute("data-url");
+    }
+  }
+
+  componentWillMount() {
+    this.props.fetchClaimListMine();
+    this._updateChannelList();
+  }
+
+  onFileChange() {
+    if (this.refs.file.getValue()) {
+      this.setState({ hasFile: true });
+    } else {
+      this.setState({ hasFile: false });
+    }
+  }
+
+  getNameBidHelpText() {
+    if (this.state.prefillDone) {
+      return (
+        <Notice>
+          {__("Existing claim data was prefilled")}
+        </Notice>
+      );
+    }
+
+    if (
+      this.state.uri &&
+      this.props.resolvingUris.indexOf(this.state.uri) !== -1 &&
+      this.claim() === undefined
+    ) {
+      return __("Checking...");
+    } else if (!this.state.name) {
+      return __("Select a URL for this publish.");
+    } else if (!this.claim()) {
+      return __("This URL is unused.");
+    } else if (this.myClaimExists() && !this.state.prefillDone) {
+      return (
+        <Notice>
```

Comment:
The `Notice` component was only used here and the style does not match the application. I dropped it.